tests/test_flows.py

```diff
@@ -0,0 +1,356 @@
+        self.debug('mac_repeater1: {}'.format(self.mac_repeater1))
+        self.mac_repeater2 = self.repeater2_ucc.dev_get_parameter('ALid')
+        self.debug('mac_repeater2: {}'.format(self.mac_repeater2))
+
+        mac_repeater1_wlan0_output = self.docker_command(self.repeater1, "ip", "-o",  "link", "list", "dev", "wlan0")
+        self.mac_repeater1_wlan0 = re.search(rb"link/ether " + RE_MAC, mac_repeater1_wlan0_output).group('mac').decode()
+        self.debug("Repeater1 wl0: {}".format(self.mac_repeater1_wlan0))
+        mac_repeater1_wlan2_output = self.docker_command(self.repeater1, "ip", "-o",  "link", "list", "dev", "wlan2")
+        self.mac_repeater1_wlan2 = re.search(rb"link/ether " + RE_MAC, mac_repeater1_wlan2_output).group('mac').decode()
+        self.debug("Repeater1 wl2: {}".format(self.mac_repeater1_wlan2))
+
+        mac_repeater2_wlan0_output = self.docker_command(self.repeater2, "ip", "-o",  "link", "list", "dev", "wlan0")
+        self.mac_repeater2_wlan0 = re.search(rb"link/ether " + RE_MAC, mac_repeater2_wlan0_output).group('mac').decode()
+        self.debug("Repeater2 wl0: {}".format(self.mac_repeater2_wlan0))
+        mac_repeater2_wlan2_output = self.docker_command(self.repeater2, "ip", "-o",  "link", "list", "dev", "wlan2")
+        self.mac_repeater2_wlan2 = re.search(rb"link/ether " + RE_MAC, mac_repeater2_wlan2_output).group('mac').decode()
+        self.debug("Repeater2 wl2: {}".format(self.mac_repeater2_wlan2))
+
+    def check_log(self, device: str, program: str, regex: str) -> bool:
+        '''Verify that on "device" the logfile for "program" matches "regex", fail if not.'''
+        logfilename = os.path.join(self.rootdir, 'logs', device, 'beerocks_{}.log'.format(program))
+        try:
+            # HACK check_log is often used immediately after triggering a message on the other side.
+            # That message needs some time to arrive on the receiver. Since our python script is pretty fast,
+            # we tend to check it too quickly. As a simple workaround, add a small sleep here.
+            # The good solution is to retry with a small timeout.
+            time.sleep(.1)
+            with open(logfilename) as logfile:
+                for line in logfile.readlines():
+                    if re.search(regex, line):
+                        self.debug("Found '{}'\n\tin {}".format(regex, logfilename))
+                        return True
+        except OSError:
+            return self.fail("Can't read {}".format(logfilename))
+        return self.fail("'{}'\n\tin log of {} on {}".format(regex, program, device))
+
+    def run_tests(self):
+        '''Run all tests as specified on the command line.'''
+        total_errors = 0
+        for test in self.opts.tests:
+            self.start_test(test)
+            self.tcpdump_start()
+            self.check_error = 0
+            try:
+                getattr(self, 'test_' + test)()
+            finally:
```

Comment:
There should be a way to find out if one of the tests in self.opts.tests does not exist. `getattr` riases an `AttributeError` in that case IIRC.

```suggestion
            except AttributeError:
                self.err("test %s does not exist" % test)
                self.check_error += 1
            finally:
```

Reply:
That's only a nitpick though, since the list of tests is generated automatically and shouldn't ever be wrong (`self.tests = [attr[len('test_'):] for attr in dir(self) if attr.startswith('test_')]`)

Reply:
Also, when it gets converted to unittest, this code will go away anyway, so no need to spend time on it I'd say.